core/src/main/scala/au/com/agiledigital/rest/security/BodyParsers.scala

```diff
@@ -11,9 +12,24 @@ import scala.concurrent.ExecutionContext
   */
 object BodyParsers {
   /**
-    * A JSON body parser that applies HTML whitelisting rules to JSON strings.
+    * A JSON body parser that applies HTML whitelisting rules to JSON strings, this removes all HTML tags.
     */
-  def whitelistingJson(implicit executionContext: ExecutionContext): BodyParser[JsValue] = parse.json map clean
+  def whitelistingJson[A](implicit executionContext: ExecutionContext, reader: Reads[A]): BodyParser[A] = parse.json map clean validate as[A]
```

Comment:
I think that we probably want to leave the variants without type parameters in place.